processing/src/main/java/org/apache/druid/timeline/CompactionState.java

```diff
@@ -58,15 +61,19 @@ public class CompactionState
   // org.apache.druid.query.aggregation.AggregatorFactory cannot be used here because it's in the 'processing' module which
   // has a dependency on the 'core' module where this class is.
   private final List<Object> metricsSpec;
+  private final CompactionEngine engine;
+  private final Map<String, AggregatorFactory> dimensionToAggregatoryFactoryMap;
 
   @JsonCreator
   public CompactionState(
       @JsonProperty("partitionsSpec") PartitionsSpec partitionsSpec,
       @JsonProperty("dimensionsSpec") DimensionsSpec dimensionsSpec,
+      @JsonProperty("dimensionToAggregatoryFactoryMap") Map<String, AggregatorFactory> dimensionToAggregatoryFactoryMap,
```

Comment:
Please annotate the new fields in the constructor and their getters as `@Nullable`.

Reply:
Reverted adding these fields.